src/lib/components/forms/form/form-components/checkbox/checkbox.component.spec.ts

```diff
@@ -1,24 +1,143 @@
+            declarations: [CheckboxComponent, MockTooltipDirective]
+        });
 
-    beforeEach(() => {
         fixture = TestBed.createComponent(CheckboxComponent);
         component = fixture.componentInstance;
+        loader = TestbedHarnessEnvironment.loader(fixture);
+
         fixture.detectChanges();
+
+        checkbox = await loader.getHarness(MatCheckboxHarness);
     });
 
-    it('should create', () => {
+    it('should create', async () => {
         expect(component).toBeTruthy();
+        expect(checkbox).toBeTruthy();
+    });
+
+    it('should disable the checkbox when #isDisabled is set', async () => {
+        expect(await checkbox.isDisabled()).toBe(false);
+
+        component.isDisabled = true;
+        fixture.detectChanges();
+
+        expect(await checkbox.isDisabled()).toBe(true);
+    });
+
+    it('should set required validation when #isRequired is set', async () => {
+        expect(await checkbox.isRequired()).toBe(false);
+
+        component.isRequired = true;
+        fixture.detectChanges();
+
+        expect(await checkbox.isRequired()).toBe(true);
+    });
+
+    it('should set #caption as label', async () => {
+        expect(await checkbox.getLabelText()).toBe('');
+
+        component.caption = 'My label';
+        fixture.detectChanges();
+
+        expect(await checkbox.getLabelText()).toBe('My label');
+    });
+
+    it('should call the callback #registerOnChange whenever the value of the checkbox is changed by the user', async () => {
+        expect(await checkbox.isChecked()).toBe(false);
+
+        const onChangeCallback: jasmine.Spy = jasmine.createSpy('onChange');
+        component.registerOnChange(onChangeCallback);
+
+        await checkbox.check();
+
+        expect(onChangeCallback).toHaveBeenCalledWith(true);
+    });
+
+    it('should call the callback #registerOnTouched whenever the checkbox was blurred', async () => {
+        const onTouchedCallback: jasmine.Spy = jasmine.createSpy('onTouched');
+        component.registerOnTouched(onTouchedCallback);
+
+        await checkbox.blur();
+
+        expect(onTouchedCallback).toHaveBeenCalled();
+    });
+
+    it('should set the value as from #writeValue', async () => {
+        const value: boolean = true;
+        component.writeValue(value);
+        expect(component.value).toEqual(value);
+
+        fixture.detectChanges();
+
+        expect(await checkbox.getValue()).toEqual(value.toString());
+    });
+
+    it('should have state inderterminate when #isIndeterminate is true', async () => {
+        expect(await checkbox.isIndeterminate()).toBe(false);
+        component.isIndeterminate = true;
+        fixture.detectChanges();
+
+        expect(await checkbox.isIndeterminate()).toBe(true);
+    });
+
+    it('should set icon as #icon', async () => {
+        let icon: MatIconHarness;
+        icon = await loader.getHarness(MatIconHarness);
+
+        expect(await icon.getName()).toBe(null);
```

Comment:
```suggestion
        expect(icon).toBe(null);
```
Ich vermute dass icon hier sogar schon null sein wird